step-api/src/main/java/org/jenkinsci/plugins/workflow/structs/DescribableHelper.java

```diff
@@ -282,9 +294,16 @@ public Type getActualType() {
         ParameterType(Type actualType) {
             this.actualType = actualType;
         }
+        
+        static ParameterType of(Type type){
+            return of(type, new Stack<String>());
+        }
 
-        static ParameterType of(Type type) {
+        static ParameterType of(Type type, Stack<String> tracker) {
             try {
+                if(tracker == null){ //in case called externally
```

Comment:
Why not just make this overload `private`? Then the `tracker` can be `@Nonnull`.